R/spatialQCPlots.R

```diff
@@ -455,11 +453,279 @@ plotZoomFovsMap <- function(spe, fovs = NULL, colour_by = NULL,
+                                       point_size=size, point_alpha=alpha)+
+                ggtitle(sample_id)+ .centroid_image_theme() + coord_fixed()
+
+            ggp2 <- scater::plotColData(spe, x=spatialCoordsNames(spe)[1],
+                                        y=spatialCoordsNames(spe)[2],
+                                        colour_by="log2AspectRatio",
+                                        point_size=size, point_alpha=alpha)+
+                ggplot2::ggtitle(sample_id) + .centroid_image_theme() +
+                ggplot2::coord_fixed()
+        }
+
+        ggp3 <- scater::plotColData(spe, x=spatialCoordsNames(spe)[1],
+                                    y=spatialCoordsNames(spe)[2],
+                                    colour_by="dist_border",
+                                    point_size=size, point_alpha=alpha)+
+            ggplot2::ggtitle(sample_id) + .centroid_image_theme() +
+            ggplot2::coord_fixed()
+
+        ggp <- cowplot::plot_grid(ggp, ggp2, ggp3, ncol = 2)
+    } else {
+        ## check if column variable is logical to impose our colors
+        ggp <- ggp1 <- scater::plotColData(spe, x=spatialCoordsNames(spe)[1],
+                                           y=spatialCoordsNames(spe)[2],
+                                           colour_by="log2CountArea",
+                                           point_size=size, point_alpha=alpha)+
+            ggplot2::ggtitle(sample_id)+ .centroid_image_theme() +
+            ggplot2::coord_fixed()
+    }
+
+    return(ggp)
+}
+
+
+
+#' FirstFilterPlot
+#' @description
+#'
+#' Plots the flagged cells identified with first filter, based on control count
+#' on total count ratio, area in um and DAPI signal.
+#'
+#' This function generates a plot that shows selected (FOVs)
+#' within a `SpatialExperiment` object, with cells flagged in different colors
+#' over a light or dark layout chosen by the user.
+#'
+#' @param spe A `SpatialExperiment` object containing spatial transcriptomics
+#' data.
+#' @param fov An integer or numeric vector specifying the FOVs to be plotted
+#' Must match values in the `fov` column of `colData(spe)`.
+#' @param theme A character string among "light" or "dark".
+#' @param custom A boolean value. If TRUE, custom polygons derived metrics will
+#' be used.
+#'
+#' @return A panel with multiple plots showing flagged cells for different
+#' variables.
+#'
+#' @importFrom dplyr case_when
+#' @importFrom ggplot2 ggplot geom_sf aes scale_fill_manual scale_color_manual
+#' @importFrom ggplot2 ggtitle theme
+#' @importFrom cowplot get_legend plot_grid
+#' @export
+#'
+#' @examples
+#' # Assuming 'spe' is a SpatialExperiment object with FOVs and polygon data:
+#' # plotZoomFovsMap(spe, fovs = c("FOV1", "FOV2"), colour_by = "cell_type",
+#' #                title = "Zoomed FOVs with Polygons")
+
+FirstFilterPlot <- function(spe, fov = unique(spe$fov), theme = c("light", "dark"), custom = FALSE){
```

Comment:
in generale non usare mai una maiuscola per il primo carattere del nome di una funzione.
Lo puoi fare solo quando scrivi il metodo di una classe, è una convenzione abbastanza comune.